### PR TITLE
Reworking material types and inheritance.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
Trying to organize things a bit better and unify all the redundant reagents with materials.
- Repathed a ton of materials under a solid/gas/liquid metal/stone/wood sort of structure.
- Merged several reagents together (chlorine, lithium, table salt, others).

*TODO*
- [x] Readd `solvent` vars.
- [x] Readd `defoliant` var.
- [x] Check for missing `dissolves_into` lists.
- [x] Double check any other removed or missed values from rebase.
- [ ] Final test and cleanup.

### CODE REVIEW

**Comment by biscuitWizard** [comment]
As long as the base root /decl/material remains /decl/material, this should be compatible with PS13.

File: code/game/machinery/kitchen/icecream.dm [hunk 1]
```diff
@@ -29,21 +29,21 @@
 /obj/machinery/icecream_vat/proc/get_ingredient_list(var/type)
 	switch(type)
 		if(ICECREAM_CHOCOLATE)
-			return list(/decl/material/chem/drink/milk, /decl/material/gas/water/ice, /decl/material/chem/nutriment/coco)
+			return list(/decl/material/chem/drink/milk, /decl/material/liquid/water, /decl/material/chem/nutriment/coco)
```
Comment (comma) [inline code/game/machinery/kitchen/icecream.dm]: mmm flavored water
maybe have machine set temp of output item to cold value?

File: code/modules/hydroponics/seed_datums.dm [hunk 2]
```diff
@@ -100,7 +100,7 @@
 	seed_name = "poison berry"
 	display_name = "poison berry bush"
 	mutants = list("deathberries")
-	chems = list(/decl/material/chem/nutriment = list(1), /decl/material/chem/toxin = list(3,5), /decl/material/chem/toxin/poisonberryjuice = list(10,5))
+	chems = list(/decl/material/chem/nutriment = list(1), /decl/material/liquid = list(3,5), /decl/material/liquid/poisonberryjuice = list(10,5))
```
Comment (comma) [inline code/modules/hydroponics/seed_datums.dm]: not toxin jsut liquid?

File: code/modules/hydroponics/seed_datums.dm [hunk 3]
```diff
@@ -113,7 +113,7 @@
 	seed_name = "death berry"
 	display_name = "death berry bush"
 	mutants = null
-	chems = list(/decl/material/chem/nutriment = list(1), /decl/material/chem/toxin = list(3,3), /decl/material/gas/carbon_monoxide = list(1,5))
+	chems = list(/decl/material/chem/nutriment = list(1), /decl/material/liquid = list(3,3), /decl/material/gas/carbon_monoxide = list(1,5))
```
Comment (comma) [inline code/modules/hydroponics/seed_datums.dm]: ditto

File: code/modules/hydroponics/trays/tray.dm [hunk 4]
```diff
@@ -49,79 +49,75 @@
...
 		/decl/material/chem/antitoxins =         -2,
-		/decl/material/chem/toxin =             2,
-		/decl/material/chem/fuel/hydrazine =         2.5,
-		/decl/material/chem/acetone =	       1,
-		/decl/material/chem/acid =              1.5,
-		/decl/material/chem/acid/hydrochloric = 1.5,
-		/decl/material/chem/acid/polyacid =     3,
-		/decl/material/chem/toxin/plantbgone =  3,
-		/decl/material/chem/radium =            2
+		/decl/material/liquid =             2,
```
Comment (comma) [inline code/modules/hydroponics/trays/tray.dm]: moar toxin -> liquid

File: code/modules/reagents/cocktails.dm [hunk 5]
```diff
@@ -17,7 +17,7 @@
 
 /decl/cocktail/proc/get_presentation_name(var/obj/item/prop)
 	. = name
-	if(prop?.reagents?.has_reagent(/decl/material/gas/water/ice) && !(/decl/material/gas/water/ice in ratios))
+	if(prop?.reagents?.has_reagent(/decl/material/liquid/water) && !(/decl/material/liquid/water in ratios))
```
Comment (comma) [inline code/modules/reagents/cocktails.dm]: gonna conflict with stuff that uses water as ingridient (like grog)

File: code/modules/reagents/reagent_containers/drinkingglass/drinkingglass.dm [hunk 6]
```diff
@@ -51,8 +51,8 @@
 /obj/item/chems/food/drinks/glass2/proc/has_ice()
 	if(LAZYLEN(reagents.reagent_volumes))
 		var/decl/material/R = reagents.get_primary_reagent_decl()
-		if(!((R.type == /decl/material/gas/water/ice) || ("ice" in R.glass_special))) // if it's not a cup of ice, and it's not already supposed to have ice in, see if the bartender's put ice in it
-			if(reagents.has_reagent(/decl/material/gas/water/ice, reagents.total_volume / 10)) // 10% ice by volume
+		if(!((R.type == /decl/material/liquid/water) || ("ice" in R.glass_special))) // if it's not a cup of ice, and it's not already supposed to have ice in, see if the bartender's put ice in it
```
Comment (comma) [inline code/modules/reagents/reagent_containers/drinkingglass/drinkingglass.dm]: maybe add a temp check

File: code/modules/reagents/reagent_containers/pill.dm [hunk 7]
```diff
@@ -91,7 +91,7 @@
 	volume = 50
 /obj/item/chems/pill/tox/Initialize()
 	. = ..()
-	reagents.add_reagent(/decl/material/chem/toxin, 50)
+	reagents.add_reagent(/decl/material/liquid, 50)
```
Comment (comma) [inline code/modules/reagents/reagent_containers/pill.dm]: another liquid toxin

File: code/modules/xenoarcheaology/artifacts/triggers/chemical.dm [hunk 8]
```diff
@@ -4,7 +4,7 @@
 
 /datum/artifact_trigger/chemical/New()
 	if(isnull(required_chemicals))
-		required_chemicals = list(pick(/decl/material/chem/acid, /decl/material/chem/toxin, /decl/material/gas/water))
+		required_chemicals = list(pick(/decl/material/liquid/acid, /decl/material/liquid, /decl/material/liquid/water))
```
Comment (comma) [inline code/modules/xenoarcheaology/artifacts/triggers/chemical.dm]: LIQUIIIIID

File: code/modules/xenoarcheaology/artifacts/triggers/chemical.dm [hunk 9]
```diff
@@ -20,31 +20,29 @@
...
 
 /datum/artifact_trigger/chemical/toxic
 	name = "presence of toxins"
 	required_chemicals = list(
-		/decl/material/chem/toxin,
-		/decl/material/chem/toxin/cyanide,
-		/decl/material/chem/toxin/amatoxin,
-		/decl/material/chem/toxin/venom,
-		/decl/material/chem/toxin/chlorine
+		/decl/material/liquid,
```
Comment (comma) [inline code/modules/xenoarcheaology/artifacts/triggers/chemical.dm]: SNAAAAKE

File: code/modules/xenoarcheaology/machinery/geosample_scanner.dm [hunk 10]
```diff
@@ -42,8 +42,8 @@
 	var/t_left_radspike = 0
 	var/rad_shield = 0
 	var/global/list/coolant_reagents_purity = list(
-		/decl/material/gas/water = 1,
-		/decl/material/gas/water/ice = 0.6,
+		/decl/material/liquid/water = 1,
```
Comment (comma) [inline code/modules/xenoarcheaology/machinery/geosample_scanner.dm:45]: double water

File: code/modules/reagents/chems/chems_medicines.dm [hunk 11]
```diff
@@ -107,11 +107,11 @@
 	var/removing = (4 * removed)
 	var/datum/reagents/ingested = M.get_ingested_reagents()
 	for(var/R in ingested.reagent_volumes)
-		if((remove_generic && ispath(R, /decl/material/chem/toxin)) || (R in remove_toxins))
+		if((remove_generic && ispath(R, /decl/material/liquid)) || (R in remove_toxins))
```
Comment (comma) [inline code/modules/reagents/chems/chems_medicines.dm]: KA NE DA (another one bit lower in diff)

**Comment by MistakeNot4892** [comment]
I readded the ice reagent since bitMuse's reagent ices need it anyway, can sort it out when I actually get far enough to add phase transitions.

Re: toxin - I have swapped it out for bromide in a few places, and made coagulated blood toxic for a couple of others. Doesn't make a huge amount of sense but generic 'toxin' no longer exists.

**Comment by MistakeNot4892** [comment]
Going to split the logic unrelated to repathing and mass code moves out of this PR into seperate ones.

File: code/modules/hydroponics/grown.dm [hunk 12]
```diff
@@ -69,33 +69,33 @@
...
+		if(reagents.has_reagent(/decl/material/liquid/nutriment/coco))
 			descriptors |= "bitter"
-		if(reagents.has_reagent(/decl/material/chem/drink/juice/orange) || reagents.has_reagent(/decl/material/chem/drink/juice/lemon) || reagents.has_reagent(/decl/material/chem/drink/juice/lime))
+		if(reagents.has_reagent(/decl/material/liquid/drink/juice/orange) || reagents.has_reagent(/decl/material/liquid/drink/juice/lemon) || reagents.has_reagent(/decl/material/liquid/drink/juice/lime))
 			descriptors |= "sweet-sour"
-		if(reagents.has_reagent(/decl/material/chem/radium) || reagents.has_reagent(/decl/material/uranium))
+		if(reagents.has_reagent(/decl/material/solid/metal/radium) || reagents.has_reagent(/decl/material/solid/metal/uranium))
 			descriptors |= "radioactive"
-		if(reagents.has_reagent(/decl/material/chem/toxin/amatoxin) || reagents.has_reagent(/decl/material/chem/toxin))
+		if(reagents.has_reagent(/decl/material/liquid/amatoxin) || reagents.has_reagent(/decl/material/liquid))
```
Comment (comma) [inline code/modules/hydroponics/grown.dm]: liquid strikes again
(also shold be property var check at this point but eh)

File: code/modules/hydroponics/seed.dm [hunk 13]
```diff
@@ -457,23 +457,23 @@
...
-		banned_chems += typesof(/decl/material/chem/nutriment)
-		banned_chems += typesof(/decl/material/chem/toxin/fertilizer)
+		banned_chems += subtypesof(/decl/material/liquid/ethanol)
+		banned_chems += subtypesof(/decl/material/solid/tobacco)
+		banned_chems += typesof(/decl/material/liquid/drink)
+		banned_chems += typesof(/decl/material/liquid/nutriment)
+		banned_chems += typesof(/decl/material/liquid/fertilizer)
 
-		if(prob(30))	banned_chems |= typesof(/decl/material/chem/toxin)
+		if(prob(30))	banned_chems |= typesof(/decl/material/liquid)
```
Comment (comma) [inline code/modules/hydroponics/seed.dm]: aaa

File: code/modules/hydroponics/seed_datums.dm [hunk 14]
```diff
@@ -100,7 +100,7 @@
 	seed_name = "poison berry"
 	display_name = "poison berry bush"
 	mutants = list("deathberries")
-	chems = list(/decl/material/chem/nutriment = list(1), /decl/material/chem/toxin = list(3,5), /decl/material/chem/toxin/poisonberryjuice = list(10,5))
+	chems = list(/decl/material/liquid/nutriment = list(1), /decl/material/liquid = list(3,5), /decl/material/liquid/poisonberryjuice = list(10,5))
```
Comment (comma) [inline code/modules/hydroponics/seed_datums.dm]: it goes it goes it goes

File: code/modules/hydroponics/seed_datums.dm [hunk 15]
```diff
@@ -113,7 +113,7 @@
 	seed_name = "death berry"
 	display_name = "death berry bush"
 	mutants = null
-	chems = list(/decl/material/chem/nutriment = list(1), /decl/material/chem/toxin = list(3,3), /decl/material/gas/carbon_monoxide = list(1,5))
+	chems = list(/decl/material/liquid/nutriment = list(1), /decl/material/liquid = list(3,3), /decl/material/gas/carbon_monoxide = list(1,5))
```
Comment (comma) [inline code/modules/hydroponics/seed_datums.dm]: liquid

File: code/modules/materials/definitions/gasses/material_gas_mundane.dm [hunk 16]
```diff
@@ -0,0 +1,274 @@
...
+	gas_specific_heat = 42.59 
+	gas_molar_mass = 0.095	  
+	gas_symbol_html = "CH<sub>3</sub>Br"
+	gas_symbol = "CH3Br"
+	taste_description = "pestkiller"
+
+/decl/material/gas/methyl_bromide/touch_turf(var/turf/T, var/amount, var/datum/reagents/holder)
+	if(istype(T))
+		var/volume = REAGENT_VOLUME(holder, type)
+		T.assume_gas(MAT_METHYL_BROMIDE, volume, T20C)
```
Comment (comma) [inline code/modules/materials/definitions/gasses/material_gas_mundane.dm]: isn't there vaporsuff for this now

Comment (MistakeNot4892) [reply code/modules/materials/definitions/gasses/material_gas_mundane.dm]: Yeah rebase artifact I think.

File: code/modules/materials/definitions/liquids/materials_liquid_solvents.dm [hunk 17]
```diff
@@ -7,33 +7,37 @@
 	touch_met = 50 // It's acid!
 	value = 1.2
 	solvent_power = MAT_SOLVENT_STRONG + 2
-	solvent_melt_dose = 10
```
Comment (comma) [inline code/modules/materials/definitions/liquids/materials_liquid_solvents.dm:10]: intended removal?

Comment (comma) [reply code/modules/materials/definitions/liquids/materials_liquid_solvents.dm:10]: in rest of file too

Comment (MistakeNot4892) [reply code/modules/materials/definitions/liquids/materials_liquid_solvents.dm:10]: Nope

File: code/modules/materials/definitions/liquids/materials_liquid_toxins.dm [hunk 18]
```diff
@@ -190,43 +150,42 @@
...
+	name = "weedkiller"
 	lore_text = "A harmful toxic mixture to kill plantlife. Do not ingest!"
 	taste_mult = 1
 	color = "#49002e"
 	toxicity = 4
-	heating_products = list(
-		/decl/material/chem/toxin = 0.3, 
-		/decl/material/gas/water = 0.7
-	)
-	defoliant = TRUE
```
Comment (comma) [inline code/modules/materials/definitions/liquids/materials_liquid_toxins.dm:213]: another effect kill

File: code/modules/reagents/reactions/reaction_solvents.dm [hunk 19]
```diff
@@ -0,0 +1,44 @@
+// TODO: solvent powers, ethanol, water
```
Comment (comma) [inline code/modules/reagents/reactions/reaction_solvents.dm]: uuuh TODO?

Comment (MistakeNot4892) [reply code/modules/reagents/reactions/reaction_solvents.dm]: This is from before I worked it into the heating/cooling section of reagent code, will remove.

**Comment by MistakeNot4892** [comment]
Far as I can tell this is now good to go. It will need better values for melting/boiling/strengthe/etc but I am happy to do that after this PR due to the current size of it.

**Review by comma — DISMISSED** [review]

**Review by afterthought2 — DISMISSED** [review]

**Review by comma — APPROVED** [review]

**Pull request merged**